src/lib/model/Async2SyncModel.js

```diff
@@ -0,0 +1,272 @@
+        } catch (err) {
+            logger.log(`Error running ${modelName} model: ${util.inspect(err)}`);
+
+            // as this function is recursive, we don't want to error the state machine multiple times
+            if (data.currentState !== 'errored') {
+                // err should not have a requestActionState property here!
+                if (err.requestActionState) {
+                    logger.log('State machine is broken');
+                }
+                // transition to errored state
+                await this.error(err);
+
+                // avoid circular ref between requestActionState.lastError and err
+                err.requestActionState = JSON.parse(JSON.stringify(this.getResponse()));
+            }
+            throw err;
+        }
+    }
+
+    const mapCurrentState = {
+        start: 'WAITING_FOR_ACTION',
+        succeeded: 'COMPLETED',
+        errored: 'ERROR_OCCURRED'
+    };
+
+    /**
+     * @name getResponse
+     * @description returns the http response payload depending on which state machine is
+     * @returns {Object} - the http response payload
+     */
+    function getResponse() {
+        const { data, logger } = this.context;
+        let resp = { ...data };
+        
+        // project some of our internal state into a more useful
+        // representation to return to the SDK API consumer
+        resp.currentState = mapCurrentState[data.currentState];
+
+        // handle unexpected state
+        if (!resp.currentState) {
+            logger.error(`${modelName} model response being returned from an unexpected state: ${data.currentState}. Returning ERROR_OCCURRED state`);
+            resp.currentState = mapCurrentState.errored;
+        }
+
+        return resp;
+    }
+    /**
+     * @name onRequestAction
+     * @description generates the pub/sub channel name
+     * @param {string} type     - the party type
+     * @param {string} id       - the party id
+     * @param {string} [subId]  - the optional party subId
+     * @returns {string} - the pub/sub channel name
+     */
+    async function onRequestAction(fsm, args) {
+        const { cache, logger } = this.context;
+        const { requests, config } = this.handlersContext;
+        logger.push({ args }).log('onRequestAction - arguments');
+        
+        return deferredJob(cache, channelNameMethod(args))
+            .init(async (channel) => {
+                const res = await requestActionMethod(requests, args);
+                logger.push({ res, channel, args }).log('RequestAction call sent to peer, listening on response');
+                return res;
+            })
+            .job((message) => {
+                this.context.data = {
+                    ...message,
+                    currentState: this.state
+                };
+                logger.push({ message }).log('requestActionMethod message received');
+            })
+            .wait(config.requestProcessingTimeoutSeconds * 1000);
```

Comment:
This is a big number if the timeout is in seconds - maybe it's meant to be in ms?

Reply:
`.requestProcessingTimeoutSeconds` the rest of the code in sdk-scheme-adapter is using this as it would be in seconds

Reply:
Yep you guys are right - I'm thinking backwards today